google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -1008,10 +1024,11 @@ public long executeUpdate(Statement update) {
     if (this.statementTag != null) {
```

Comment:
This `if` statement causes the priority value to only be set if there is also a statement tag. If the statement tag is `null` and the client has set a custom priority, the connection will still just use the default priority.

I also think that if you want to reuse this method for also setting the priority, that you should rename the method to better reflect what it is doing.

One thing that should also be considered here is that this method (and the one for update statements) are only invoked for queries and updates. They are not called for `Commit` invocations (and also not for `Rollback`, but that is probably less of an issue).